app/src/main/java/org/simple/clinic/registration/phone/RegistrationPhoneModel.kt

```diff
@@ -2,20 +2,25 @@ package org.simple.clinic.registration.phone
 
 import android.os.Parcelable
 import kotlinx.android.parcel.Parcelize
+import org.simple.clinic.user.OngoingRegistrationEntry
 
 @Parcelize
 data class RegistrationPhoneModel(
-    private val enteredPhoneNumber: String
+    private val ongoingRegistrationEntry: OngoingRegistrationEntry?
 ) : Parcelable {
 
   companion object {
 
     fun create(): RegistrationPhoneModel = RegistrationPhoneModel(
-        enteredPhoneNumber = ""
+        ongoingRegistrationEntry = null
     )
   }
 
   fun phoneNumberChanged(phoneNumber: String): RegistrationPhoneModel {
-    return copy(enteredPhoneNumber = phoneNumber)
+    // TODO (vs) 04/06/20: Change in a later commit to not require a null check
+    return if (ongoingRegistrationEntry != null)
+      copy(ongoingRegistrationEntry = ongoingRegistrationEntry.withPhoneNumber(phoneNumber))
+    else
+      this
```

Comment:
We could. In a later part of the migration, this check has been removed since the ongoing entry is no longer nullable. So this change would be redundant.